src/citrine/resources/material_spec.py

```diff
@@ -105,12 +112,43 @@ def filter_by_template(self,
         :param per_page: The number of results to return per page.
         :return: A search result of material specs
         """
-        path_prefix = MaterialTemplateCollection(self.project_id,
-                                                 self.dataset_id,
-                                                 self.session)._get_path(ignore_dataset=True)
-        path = path_prefix + "/" + template_scope + "/" + template_id + "/material-specs"
-        raw_objects = self.session.cursor_paged_resource(self.session.get_resource,
-                                                         path,
-                                                         per_page=per_page,
-                                                         version="v1")
-        return (self.build(raw) for raw in raw_objects)
+        if per_page is not None:
+            logger.warning('The per_page parameter will be ignored. Please remove it.')
+        return self.list_by_template(uid=template_id, scope=template_scope)
+
+    def list_by_template(self, uid: Union[UUID, str], scope: str = 'id') -> Iterator[MaterialSpec]:
+        """
+        [ALPHA] Get the material specs using the specified material template.
+
+        Parameters
+        ----------
+        uid
+            The unique ID of the material template whose material spec usages are to be located.
+        scope
+            The scope of `uid`.
+        Returns
+        -------
+        Iterator[MaterialSpec]
+            The material specs using the specified material template.
+
+        """
+        return self._get_relation('material-templates', uid=uid, scope=scope)
+
+    def get_by_process(self, uid: Union[UUID, str], scope: str = 'id') -> Optional[MaterialSpec]:
```

Comment:
should this be called list_by_process?

Reply:
Validations ensure that there is at most 1 material associated with a process.